packages/build/src/plugins_core/blobs_upload/index.ts

```diff
@@ -0,0 +1,84 @@
+import { version as nodeVersion } from 'node:process'
+
+import { getDeployStore } from '@netlify/blobs'
+import pMap from 'p-map'
+import semver from 'semver'
+
+import { addErrorInfo } from '../../error/info.js'
+import { log, logError } from '../../log/logger.js'
+import { getBlobsDir } from '../../utils/blobs.js'
+
+import { getKeysToUpload, getFileWithMetadata, anyBlobsToUpload } from './utils.js'
+
+const coreStep = async function ({
+  debug,
+  logs,
+  deployId,
+  buildDir,
+  quiet,
+  constants: { PUBLISH_DIR, SITE_ID, NETLIFY_API_TOKEN, API_URL },
+}) {
+  const storeOpts: { siteID: string; deployID: string; token: string; apiURL: string; fetch?: any } = {
+    siteID: SITE_ID,
+    deployID: deployId,
+    token: NETLIFY_API_TOKEN,
+    apiURL: API_URL,
+  }
+  if (semver.lt(nodeVersion, '18.0.0')) {
```

Comment:
nice!

Reply:
We will be calling this step from the CLI when doing `netlify deploy` (with or without `--build`).
However this version check is needed because we still run tests with Node 14 in CI here because we still support Node 14/16 https://github.com/netlify/build/blob/main/packages/build/package.json#L155
CLI already requires Node 18+, and in buildbot we run Node 18 as well for this piece. However, I didn't want to block this work on updating requirements for build. We can do that as a follow-up.